utils/device_info_generator.py

```diff
@@ -101,19 +109,7 @@ def ioc_name(self, auto=False):
 
         Returns: The name of the IOC based on the input name. Must be between 1 and 8 characters
         """
-        if self._ioc_name is None:
-            default_ioc_name = self._upper_case_spaces_removed_no_truncation()
-            if self._is_valid_ioc_name(default_ioc_name):
-                proposed_name = default_ioc_name
-            elif auto:
-                proposed_name = self._auto_generated_ioc_name()
-            else:
-                proposed_name = default_ioc_name
-                while not self._is_valid_ioc_name(proposed_name):
-                    proposed_name = get_input(
-                        "Device name, {}, is invalid. Please enter a valid IOC name: ".format(proposed_name))
-            self._ioc_name = proposed_name
-        return self._ioc_name
+        return self._upper_case_spaces_removed_no_truncation()
```

Comment:
Just some tidying up:

`_auto_generated_ioc_name()` is now unused. I don't see it providing much value over `_upper_case_spaces_removed_no_truncation()` - if you agree can you please remove the redundant method? Otherwise please add a validity check for `_auto_generated_ioc_name()` in the constructor before the user prompt. 

Also please remove the now unused `auto` parameter.